iroha/src/asset.rs

```diff
@@ -6,101 +6,103 @@ use parity_scale_codec::{Decode, Encode};
 }
 
 impl Asset {
     pub fn new(id: Id) -> Self {
-        Asset { id, amount: 0 }
+        Asset { id, quantity: 0 }
     }
 
-    pub fn with_amount(mut self, amount: u128) -> Self {
-        self.amount = amount;
+    pub fn with_quantity(mut self, quantity: u128) -> Self {
+        self.quantity = quantity;
         self
     }
 }
 
-pub mod isi {
-    use super::*;
-    use crate::isi::Contract;
-    use iroha_derive::{IntoContract, Io};
-    use parity_scale_codec::{Decode, Encode};
+/// Identification of an Asset. Consists of Asset's name, Domain's name and Account's name.
+///
+/// # Example
+///
+/// ```
+/// use iroha::asset::Id;
+///
+/// let id = Id::new("xor", "user", "company");
+/// ```
+#[derive(Clone, Debug, PartialEq, PartialOrd, Ord, Eq, std::hash::Hash, Encode, Decode)]
+pub struct Id {
+    pub name: String,
+    pub container: String,
+    pub account: String,
+}
 
-    /// The purpose of add asset quantity command is to increase the quantity of an asset on account of
-    /// transaction creator. Use case scenario is to increase the number of a mutable asset in the
-    /// system, which can act as a claim on a commodity (e.g. money, gold, etc.).
-    #[derive(Clone, Debug, PartialEq, Io, IntoContract, Encode, Decode)]
-    pub struct AddAssetQuantity {
-        pub asset_id: Id,
-        pub account_id: Id,
-        pub amount: u128,
+impl Id {
+    pub fn new(name: &str, container: &str, account: &str) -> Self {
+        Id {
+            name: name.to_string(),
+            container: container.to_string(),
+            account: account.to_string(),
+        }
     }
 
-    impl Instruction for AddAssetQuantity {
-        fn execute(&self, world_state_view: &mut WorldStateView) -> Result<(), String> {
-            let assets = &mut world_state_view
-                .account(&self.account_id)
-                .ok_or("Account not found.")?
-                .assets;
-            if let Some(asset) = assets.get_mut(&self.asset_id) {
-                asset.amount += self.amount;
-            } else {
-                assets.insert(self.asset_id.clone(), Asset::new(self.asset_id.clone()));
-                assets
-                    .get_mut(&self.asset_id)
-                    .ok_or("Failed to get asset.")?
-                    .amount += self.amount;
-            }
-            Ok(())
-        }
+    pub fn account_id(&self) -> <Account as Identifiable>::Id {
+        <Account as Identifiable>::Id::new(&self.account, &self.container)
     }
+}
 
-    /// The purpose of сreate asset command is to create a new type of asset, unique in a domain.
-    /// An asset is a countable representation of a commodity.
-    #[derive(Clone, Debug, PartialEq, Io, IntoContract, Encode, Decode)]
-    pub struct CreateAsset {
-        pub asset_name: String,
-        pub domain_id: String,
-        pub decimals: u8,
+impl From<&str> for Id {
+    fn from(string: &str) -> Id {
+        let vector: Vec<&str> = string.split('@').collect();
```

Comment:
assets don't use the @ symbol, but rather #. Domain#asset_name is the standard.

Reply:
https://soramitsu.atlassian.net/browse/RH2-88?atlOrigin=eyJpIjoiMWY2NmQ2NDVhNjc3NGE5NWIyNGQ2MDY3NTIwOTQwOTciLCJwIjoiaiJ9